backend/server.js

```diff
@@ -18,12 +44,54 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
+const authenticateUser = async (req, res, next) => {
+  const user = await User.findOne({ accessToken: req.header('Authorization') })
+  if (user) {
+    req.user = user //what does this mean? 
+    next() //when to use next? (calling the next() function which allows the proteced endpoint to continue execution)
```

Comment:
the next() is used in these middle wares, as the one I pasted in above and also here when you want to restict if to go on to next function or not. I think Damien explains this in the video of error handling.